src/shopify/theme_client.go

```diff
@@ -280,14 +280,18 @@ func (c Client) GetAsset(filename string) (Asset, error) {
 // If there was an error, in the request then error will be defined otherwise the
 // response will have the appropriate data for usage.
 func (c Client) CreateAsset(asset Asset) error {
-	return c.UpdateAsset(asset)
+	return c.UpdateAsset(asset, "")
 }
 
-// UpdateAsset will take an asset and will return  when the asset has been updated.
+// UpdateAsset will take an asset and will return when the asset has been updated.
 // If there was an error, in the request then error will be defined otherwise the
 // response will have the appropriate data for usage.
-func (c Client) UpdateAsset(asset Asset) error {
-	resp, err := c.http.Put(c.assetPath(map[string]string{}), map[string]Asset{"asset": asset}, nil)
+func (c Client) UpdateAsset(asset Asset, prevChecksum string) error {
+	var header = make(map[string]string)
+	if prevChecksum != "" {
+		header["X-Shopify-Previous-Checksum"] = prevChecksum
```

Comment:
@attaboy @gfscott 

@tanema suggested getting your thoughts on what to call this header (`X-Shopify-Something`).

It's for detecting concurrency conflicts, and it somewhat similar to an [Etag](https://en.wikipedia.org/wiki/HTTP_ETag).

When`theme watch` is started, we grab the checksums of the assets on the server. When an asset is updated, we send the old checksum as part of the `PUT`. We can then determine if the asset was changed another user, and avoid accidental overwrites.

'previous' seems a little vague... any thoughts on something clearer?

Reply:
Interesting! Agree that `previous` isn't quite specific enough.

What about `X-Shopify-Latest-Checksum` — seems like it more accurately identifies that this is "the single most recent checksum" (as opposed to "any previous checksum"). BUT does it risk being confused with the _new_ checksum of the updated asset?

Reply:
`X-Shopify-Last-Known-Checksum` maybe?

Reply:
or `X-Shopify-Checksum-On-Download`

Reply:
I like `Last-Known-Checksum`, I think it's properly specific and unambiguous.

Reply:
👍 Updated in b5e12a5

Reply:
@gfscott, @tanema suggested using something closer to the `If-Match` convention used be ETags:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag

We're now thinking `Replace-If-Checksum-Match`, which is like saying to the server, "I'm giving this updated asset to you, if the checksum it's signed with matches your curret checksum, then replace it, otherwise reject it".

Reply:
Cool! if there's a recognizable pattern we can lean on, that's preferable for sure.

Does the development handbook have general guidelines on naming for things like http headers? I did [a quick search](https://development.shopify.io/search?q=naming) initially and didn't find much.